src/api/spec/controllers/staging/staged_requests_controller_spec.rb

```diff
@@ -418,6 +419,21 @@
         it { expect(staging_project.packages).to be_empty }
         it { expect(staging_project.staged_requests).to be_empty }
       end
+
+      context 'with declined request' do
+        before do
+          login user
+          bs_request.change_state(newstate: 'declined', user: user.login, comment: 'Fake comment')
+          delete :destroy, params: { staging_workflow_project: staging_workflow.project.name, format: :xml },
+                           body: "<requests><request id='#{bs_request.number}'/></requests>"
```

Comment:
see the "test case" in the feature request. The interesting part is if reopening the request leaves the staging project untouched and the request will appear in backlog.

I expect this to be the case, but it would be good to have this future safe.

Reply:
@coolo, what do you mean with untouched?
I extended the test, reopening a declined request that was unstaged.

Reply:
untouched == it stays empty after reopen if it was empty before. Just as your test verifies